velox/functions/lib/aggregates/SingleValueAccumulator.cpp

```diff
@@ -49,15 +49,30 @@ bool SingleValueAccumulator::hasValue() const {
   return start_.header != nullptr;
 }
 
-int32_t SingleValueAccumulator::compare(
+std::optional<int32_t> SingleValueAccumulator::compare(
     const DecodedVector& decoded,
-    vector_size_t index) const {
+    vector_size_t index,
+    CompareFlags compareFlags) const {
   VELOX_CHECK_NOT_NULL(start_.header);
-
   ByteStream stream;
   HashStringAllocator::prepareRead(start_.header, stream);
-  return exec::ContainerRowSerde::compare(
-      stream, decoded, index, {true, true, false});
+  auto result = exec::ContainerRowSerde::compareWithNulls(
+      stream, decoded, index, compareFlags);
+
+  VELOX_USER_CHECK(result.has_value());
```

Comment:
According to the comment in the header file, this method can return std::nullopt, right?

Reply:
Yes, it can. This is not right here, we should return the result directly.
This PR was asked to be split into 2 PRs, so I fixed it in a new PR https://github.com/facebookincubator/velox/pull/6570.